### ISSUE TITLE
Inline chat - Single edit mode

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#28792]
- Utilizing the Inline chat as means for showing edits.

TLDR; The Inline chat could be cheaply extended to provide In editor edits. (Happy to provide the PR)

![image](https://github-com/microsoft/vscode/assets/13024177/a3ab657b-5c42-4ac9-94fa-602622819c36)

-- Intro

At the moment the experimental inline chat, provides very interesting way how to display edits that are restricted to a single file (TextEdits). And assuming that this API is something that the extension authors are interested in (e.g. see https://github-com/microsoft/vscode/-/85682 ) might provide a great addition to the extension author toolbelt.

The main benefits being (compared to Refactor Preview), that for small edits/single file edits:

- the user focus is not broken by providing unnecessarily big context
- suggested edits are editable
- Reduces the amount of clicks the user needs, to view edits

At the same time, I realise that the Inline chat purpose is to **Chat**, but at the moment it seems that the edit preview logic is fused with the internals of the chat and can not be easily extracted.

-- Proposal

Extend the existing `inlineChat.start` (or introduce a new command e.g. `inlineChat.showEdit`) which would allow to take in `edits: vscode.TextEdit[]` (or vscode.WorkspaceEdit, and add additional checks in the inlineChat). The edits would be added to the options (which are passed to the `InlineChatController`), and would modify the state flow a bit.
After `INIT_UI`, break the `_nextState` and Attach the edit as a new response, then continue the `_nextState` with `APPLY_EDIT` 
If edits are provided in the options, then hide parts of the toolbar (the conversational pieces), and keep only the actions.

--- Additional nice to have

- Allow to accept/reject the edit via command/api invocation from extension (only for the extension provided provider)
- Ability to add Markdown info text inside the Action bar (e.g. to provide some helpful hint for the edit)

VS Code version: 1.81.1 
OS version: ChromeOS  114.0.5735.239
